controllers/machinehealthcheck_controller.go

```diff
@@ -192,3 +265,156 @@ func (r *MachineHealthCheckReconciler) clusterToMachineHealthCheck(o handler.Map
 	}
 	return requests
 }
+
+func namespacedName(obj metav1.Object) types.NamespacedName {
+	return types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()}
+}
+
+// machineToMachineHealthCheck maps events from Machine objects to
+// MachineHealthCheck objects that monitor the given machine
+func (r *MachineHealthCheckReconciler) machineToMachineHealthCheck(o handler.MapObject) []reconcile.Request {
+	m, ok := o.Object.(*clusterv1.Machine)
+	if !ok {
+		r.Log.Error(errors.New("incorrect type"), "expected a Machine", "type", fmt.Sprintf("%T", o))
+		return nil
+	}
+
+	mhcList := &clusterv1.MachineHealthCheckList{}
+	if err := r.Client.List(
+		context.Background(),
+		mhcList,
+		&client.ListOptions{Namespace: m.Namespace},
+		client.MatchingFields{mhcClusterNameIndex: m.Spec.ClusterName},
+	); err != nil {
+		r.Log.Error(err, "Unable to list MachineHealthChecks", "machine", m.Name, "namespace", m.Namespace)
+		return nil
+	}
+
+	var requests []reconcile.Request
+	for k := range mhcList.Items {
+		if r.hasMatchingLabels(&mhcList.Items[k], m) {
+			key := types.NamespacedName{Namespace: mhcList.Items[k].Namespace, Name: mhcList.Items[k].Name}
+			requests = append(requests, reconcile.Request{NamespacedName: key})
+		}
+	}
+	return requests
+}
+
+func (r *MachineHealthCheckReconciler) nodeToMachineHealthCheck(o handler.MapObject) []reconcile.Request {
+	node, ok := o.Object.(*corev1.Node)
+	if !ok {
+		r.Log.Error(errors.New("incorrect type"), "expected a Node", "type", fmt.Sprintf("%T", o))
+		return nil
+	}
+
+	machine, err := r.getMachineFromNode(node.Name)
+	if machine == nil || err != nil {
+		r.Log.Error(err, "Unable to retrieve machine from node", "node", namespacedName(node).String())
```

Comment:
nit: `namespacedName()` is actually only being used here. We can probably drop it and just `node.GetName()` here.